### PR TITLE
Update and_gate.py

### PR DESCRIPTION
**Pull request by hritik1804**
### Describe your change:

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: boolean_algebra/and_gate.py [hunk 1]
```diff
@@ -28,7 +28,11 @@ def and_gate(input_1: int, input_2: int) -> int:
     0
     >>> and_gate(1, 1)
     1
+    >>> and_gate(2, 1)  # Added test case with invalid input
+    "Please enter valid input values: 0 or 1"
     """
+    if input_1 not in (0, 1) or input_2 not in (0, 1):
```
Comment (CaedenPH) [inline boolean_algebra/and_gate.py:34]:
```suggestion
    if input_1 not in {0, 1} or input_2 not in {0, 1}:
```

File: boolean_algebra/and_gate.py [hunk 2]
```diff
@@ -28,7 +28,11 @@ def and_gate(input_1: int, input_2: int) -> int:
...
+        return "Please enter valid input values: 0 or 1"
```
Comment (CaedenPH) [inline boolean_algebra/and_gate.py:35]: This should raise an error instead

Comment (hritik1804) [reply boolean_algebra/and_gate.py:35]: There is no errors raises i got a successful output while executing it

**Comment by CaedenPH** [comment]
@cclauss Just curious why you're closing pull requests without comments explaining

**Comment by cclauss** [comment]
Perhaps because less than a week ago we had zero open pull requests and today we have more than 85 and we have been working hard to merge useful ones for all that time.

Hacktoberfest blows!  This is a voluntary activity and we are trying to keep quality and relevance high.

**Pull request closed without merging**